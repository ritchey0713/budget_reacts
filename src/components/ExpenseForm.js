import React from "react"

export default class ExpenseForm extends React.Component {
  render(){
    return (
      <div>
        Expense form
      </div>
    )
  }
}