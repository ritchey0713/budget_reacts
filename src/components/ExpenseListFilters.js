import React from 'react'
import { connect } from "react-redux"
import {setTextFilter, sortByDate, sortByAmount, setStartDate, setEndDate} from "../actions/filters"
import { DateRangePicker } from 'react-dates'

class ExpenseListFilters extends React.Component {
  state = {
    calanderFocused: null,

  }

  onDatesChange = ({ startDate, endDate }) => {
    this.props.dispatch(setStartDate(startDate))
    this.props.dispatch(setEndDate(endDate))
  }

  onFocusChange = (calanderFocused) => {
    this.setState(() => {
      return {
        calanderFocused: calanderFocused
      }
    })
  }

  render(){
    return (
      (
        <div className="content-container">
          <div className="input-group">
            <div className="input-group__item">
              <input 
                type="text" 
                className="text-input"
                value={this.props.filters.text} 
                onChange={(e) => {
                  this.props.dispatch(setTextFilter(e.target.value))
                  }
                }
              />
            </div>  
            <div className="input-group__item">
              <select 
                className="select"
                value={this.props.filters.sortBy} 
                onChange={(e) => {
                  e.target.value === "amount" ? this.props.dispatch(sortByAmount()) : this.props.dispatch(sortByDate())
                  }
                }
              >
                <option value="date">Date</option>
                <option value="amount">Amount</option>
              </select>
            </div>  
            <div className="input-group__item">
              <DateRangePicker 
                startDate={this.props.filters.startDate}
                startDateId={Math.random().toString()}
                endDate={this.props.filters.endDate}
                endDateId={Math.random().toString()}
                onDatesChange={this.onDatesChange}
                focusedInput={this.state.calanderFocused}
                onFocusChange={this.onFocusChange}
                numberOfMonths={1}
                isOutsideRange={() => false}
                showClearDates={true}
              />
            </div>  
          </div>
        </div>
      )
    )
  }
}



const mapStateToProps = (state, ownProps) => {
  return {
    filters: state.filters
  }
}

export default connect(mapStateToProps)(ExpenseListFilters)