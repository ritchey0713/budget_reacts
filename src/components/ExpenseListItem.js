import React from "react"
import PropTypes from "prop-types"
import { removeExpense } from "../actions/expenses"
import { connect } from 'react-redux'
// const ExpenseListItem = (props) => (
//   <div>
//     <h3>
//       DESC: {props.expense.description}  
//     </h3>

//     <p>
//       Created at: {props.expense.createdAt}  
//     </p>
//   </div>
// )

const ExpenseListItem = ({ id, description, amount, createdAt, dispatch }) => (
  <div>
    <h3>
      DESC: {description}  
    </h3>

    <p>
      Amount: {amount}  
    </p>

    <p>
      Created at: {createdAt}  
    </p>

    <button onClick={(e) => {
      dispatch(removeExpense({ id }))
      }}>
      remove
    </button>
  </div>
)

ExpenseListItem.propTypes = {
  description: PropTypes.string.isRequired,
  amount: PropTypes.number.isRequired,
  createdAt: PropTypes.string.isRequired
}



export default connect()(ExpenseListItem)