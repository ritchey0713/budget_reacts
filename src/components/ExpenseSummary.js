import React from "react"
import { connect } from 'react-redux'
import numeral from 'numeral'
import selectExpenses from "../selectors/expenses"
import selectExpensesTotal from "../selectors/expenses-total"

export const ExpensesSummary = (props) => {
  const expenseWord = props.expensesCount === 1 ? "expense" : "expenses";
  const formattedExpensesTotal = numeral(props.expensesTotal / 100).format("$0,0.00");
  console.log(formattedExpensesTotal)
  
  return (
    <div>
      <h1>Viewing {props.expensesCount} {expenseWord} totalling {formattedExpensesTotal} </h1>
    </div>
  )
}

const mapStateToProps = (state, ownProps) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters)
  return {
    expensesCount : visibleExpenses.length,
    expensesTotal: selectExpensesTotal(visibleExpenses)
  }
}

export default connect(mapStateToProps)(ExpensesSummary)