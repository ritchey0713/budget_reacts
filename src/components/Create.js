import React from "react" 
import ExpenseForm from "./ExpenseForm"
import { connect } from 'react-redux'
import { addExpense } from "../actions/expenses"

const CreateExpense = (props) => {
  return (
    <div>
      <h1>Add new budget</h1>
      <ExpenseForm
        reduxOnSubmit={(expense) => {
          console.log(expense)
          props.dispatch(addExpense(expense))
          props.history.push("/")
        }}
      />
    </div>
  )
}

export default connect()(CreateExpense)