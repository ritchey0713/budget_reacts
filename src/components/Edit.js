import React from 'react'
import ExpenseForm from "./ExpenseForm"
import { connect } from 'react-redux'
import { startEditExpense, startRemoveExpense } from '../actions/expenses'

const EditExpense = (props) => {
  return (
    <div>
      <div className="page-header">
        <div className="content-container">
        <h1 className="page-header__title">Editing: {props.expense.description}</h1>
        </div>
      </div>
      <div className="content-container">
        <ExpenseForm 
          expense={props.expense}
          reduxOnSubmit={(expense) => {
            props.dispatch(startEditExpense(props.expense.id, expense))
            props.history.push("/")
            }
          }
        />
        <button 
          className="button button--secondary"
          onClick={(e) => {
          props.dispatch(startRemoveExpense(props.expense))
          props.history.push("/")
          }}>
          Remove
        </button>
      </div>

      
    </div>
  )
}

const mapStateToProps = (state, ownProps) => {
  return {
    expense: state.expenses.find((expense) => {
      return expense.id === ownProps.match.params.id
    })
  }
}

export default connect(mapStateToProps)(EditExpense)