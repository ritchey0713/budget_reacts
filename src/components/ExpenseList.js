import React from "react"
import PropTypes from "prop-types"
import { connect } from 'react-redux'
import ExpenseListItem from "./ExpenseListItem"
import selectExpenses from "../selectors/expenses"
const ExpenseList = (props) => {
  return (
    <div className="content-container">
      <div className="list-header">
        <div className="show-for-mobile">Expenses</div>
        <div className="show-for-desktop">Expense</div>
        <div className="show-for-desktop">Amount</div>
      </div>
      <div className="list-body">
        {props.expenses.length === 0 ? (
          <div className="list-item list-item--message">
            <span>No Expenses</span>
          </div>
          ) : (
            props.expenses.map((expense) => {
              return (
                <ExpenseListItem
                  {...expense}
                  key={expense.id || "test-key"}
                />
          
          ) 
        }))}
      </div>
    </div>
  )
}

const mapStateToProps = (state, ownProps) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  }
}

ExpenseList.propTypes = {
  expenses: PropTypes.array.isRequired,
}

export default connect(mapStateToProps)(ExpenseList)
