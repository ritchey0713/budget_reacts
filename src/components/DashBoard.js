import React from "react"
import ExpenseList from "./ExpenseList"
import ExpenseListFilters from "./ExpenseListFilters"
import ExpenseSummary from "./ExpenseSummary"

const dashBoard = () => {
  return (
    <div>
      <ExpenseSummary />
      <ExpenseListFilters/>
      <ExpenseList />
    </div>
  )
}

export default dashBoard