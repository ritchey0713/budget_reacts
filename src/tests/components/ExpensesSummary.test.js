import React from "react"
import { shallow } from "enzyme"
import { ExpensesSummary } from "../../components/ExpenseSummary"

test("should render summary with one expense", () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={500} />)
  expect(wrapper).toMatchSnapshot()
})

test("should render summary with multiple expenses", () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={15} expensesTotal={5002323232}  />)
  expect(wrapper).toMatchSnapshot()
})