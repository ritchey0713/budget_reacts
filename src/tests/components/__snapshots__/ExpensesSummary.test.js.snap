// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render summary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        15
      </span>
       
      expenses
       totalling 
      <span>
        $50,023,232.32
      </span>
       
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should render summary with one expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $5.00
      </span>
       
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
